pkg/kubelet/dockertools/kube_docker_client.go

```diff
@@ -146,56 +146,41 @@ func (d *kubeDockerClient) RemoveContainer(id string, opts dockertypes.Container
 	return d.client.ContainerRemove(getDefaultContext(), opts)
 }
 
-func (d *kubeDockerClient) InspectImage(image string) (*docker.Image, error) {
+func (d *kubeDockerClient) InspectImage(image string) (*dockertypes.ImageInspect, error) {
 	resp, _, err := d.client.ImageInspectWithRaw(getDefaultContext(), image, true)
 	if err != nil {
-		// TODO(random-liu): Use IsErrImageNotFound instead of ErrNoSuchImage
 		if dockerapi.IsErrImageNotFound(err) {
-			err = docker.ErrNoSuchImage
+			err = imageNotFoundError{ID: image}
 		}
 		return nil, err
 	}
-	imageInfo := &docker.Image{}
-	if err := convertType(&resp, imageInfo); err != nil {
-		return nil, err
-	}
-	return imageInfo, nil
+	return &resp, nil
 }
 
-func (d *kubeDockerClient) ListImages(opts docker.ListImagesOptions) ([]docker.APIImages, error) {
-	resp, err := d.client.ImageList(getDefaultContext(), dockertypes.ImageListOptions{
-		MatchName: opts.Filter,
-		All:       opts.All,
-		Filters:   convertFilters(opts.Filters),
-	})
+func (d *kubeDockerClient) ListImages(opts dockertypes.ImageListOptions) ([]dockertypes.Image, error) {
+	images, err := d.client.ImageList(getDefaultContext(), opts)
 	if err != nil {
 		return nil, err
 	}
-	images := []docker.APIImages{}
-	if err = convertType(&resp, &images); err != nil {
-		return nil, err
-	}
 	return images, nil
 }
 
-func base64EncodeAuth(auth docker.AuthConfiguration) (string, error) {
+func base64EncodeAuth(auth dockertypes.AuthConfig) (string, error) {
 	var buf bytes.Buffer
 	if err := json.NewEncoder(&buf).Encode(auth); err != nil {
 		return "", err
 	}
 	return base64.URLEncoding.EncodeToString(buf.Bytes()), nil
 }
 
-func (d *kubeDockerClient) PullImage(opts docker.PullImageOptions, auth docker.AuthConfiguration) error {
+func (d *kubeDockerClient) PullImage(image string, auth dockertypes.AuthConfig, opts dockertypes.ImagePullOptions) error {
+	// RegistryAuth is the base64 encoded credentials for the registry
 	base64Auth, err := base64EncodeAuth(auth)
 	if err != nil {
 		return err
 	}
-	resp, err := d.client.ImagePull(getDefaultContext(), dockertypes.ImagePullOptions{
-		ImageID:      opts.Repository,
-		Tag:          opts.Tag,
-		RegistryAuth: base64Auth,
-	}, nil)
+	opts.RegistryAuth = base64Auth
+	resp, err := d.client.ImagePull(getDefaultContext(), opts, nil)
```

Comment:
@resouer Could you fix this? After this we could get this PR merged as soon as possible. :)